example/postconstructorpredestructor/app.d

```diff
@@ -42,8 +42,14 @@ class AClass {
 
 public void main() {
 	auto container = new shared DependencyContainer();
-	container.register!ADependency;
-	container.register!AClass;
+	container.register!(ADependency).onConstructed((Object obj) {
+		writeln("ADependency constructed");

```

Comment:
I use poodinis for my project and would rather see use of the standard logger than a writeln. For support across compiler versions it should be wrapped, something like:

```d
    static if(__traits(compiles, (){ import std.experimental.logger; } )) {
        sharedLog.info("ADependency contructed");
    }
```

doing this way allows end users to have control over whether they see the message based on log level and also allows it to be logged to various output

Reply:
Yes, we also recommend that the logger should be used to replace the simple writeln().
That seems another PR needed .

Reply:
These writelns are currently in examples. Examples should be kept as simple as possible. There's no need to use the logger API there, that's not the goal of the example.

Reply:
These writelns are currently in examples. Examples should be kept as simple as possible. There's no need to use the logger API there, that's not the goal of the example.

Reply:
that's fine for this file but there's similar in _source/poodinis/container.d_ I chose bad place to put the comment but meant as a general principal. It's good if logging is handled by something like:

```d
    static if(__traits(compiles, (){ import std.experimental.logger; } )) {
        sharedLog.info("blah blah blah");
    } else if(CHECK FOR VIBE-D) {
        // use vibe log
    } else {
        writeln("blah blah blah");
    }
```

as it allows for your end users to decide how they want log output.